src/ForceTorqueMonitor.c

```diff
@@ -0,0 +1,164 @@
+// ForceTorqueMonitor.c
+
+// SPDX-FileCopyrightText: 2022-2023, Yaskawa America, Inc.
+// SPDX-FileCopyrightText: 2022-2023, Delft University of Technology
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#include "MotoROS.h"
+
+ForceTorqueMonitor_Publishers g_publishers_ForceTorqueMonitor;
+ForceTorqueMonitor_Messages g_messages_ForceTorqueMonitor;
+
+static void Ros_ForceTorqueMonitor_Initialize_Publishers(rmw_qos_profile_t const* const qos_profile);
+
+void Ros_ForceTorqueMonitor_Initialize()
+{
+    Ros_Debug_BroadcastMsg("Initializing ForceTorqueMonitor publishers");
+
+    //==================================
+    //create publishers for TCP force and external joint torque
+    const rmw_qos_profile_t* qos_profile_js = Ros_ConfigFile_To_Rmw_Qos_Profile(g_nodeConfigSettings.qos_joint_states);
+    Ros_ForceTorqueMonitor_Initialize_Publishers(qos_profile_js);
+
+}
+
+static void Ros_ForceTorqueMonitor_Initialize_Publishers(rmw_qos_profile_t const* const qos_profile)
+{
+    //create a publisher for TCP force torque
+    rcl_ret_t ret = rclc_publisher_init(
+        &g_publishers_ForceTorqueMonitor.forceTorqueTCP,
+        &g_microRosNodeInfo.node,
+        ROSIDL_GET_MSG_TYPE_SUPPORT(geometry_msgs, msg, WrenchStamped),
+        TOPIC_NAME_FORCE_TORQUE_TCP,
+        qos_profile);
+    motoRosAssert(ret == RCL_RET_OK, SUBCODE_FAIL_CREATE_PUBLISHER_JOINT_STATE_ALL);
```

Comment:
I like to have a unique error code for every single alarm. That way, the numerical code identifies the exact line line of code which raised the alarm. There's no speculation about possible causes.

Please add a new entry to the end of the `ALARM_ASSERTION_FAIL_SUBCODE` enum. (If you add to the middle of the enum, then it throws off the existing troubleshooting documentation.) Then we'll also need to add this entry to `troubleshooting.md`.